src/main/java/software/amazon/event/kafkaconnector/EventBridgeSinkTask.java

```diff
@@ -59,102 +70,100 @@ public void put(Collection<SinkRecord> records) {
-            .filter(Objects::nonNull)
-            .collect(Collectors.toList());
-
-    // TODO: disabling batching for now until we can correctly calculate the size of the whole batch
-    // to avoid failing all entries in an EventBridge request due to exceeding 256kb size limit
-    var batches = Lists.partition(eventBridgeWriterRecords, 1);
+    var remainingRecords = List.copyOf(records);
 
-    var attempts = 0;
+    var attempts = new AtomicInteger(0);
     var start = OffsetDateTime.now(ZoneOffset.UTC);
-    while (true) {
-      attempts++;
+
+    while (!remainingRecords.isEmpty()) {
+      attempts.incrementAndGet();
       log.trace(
           "EventBridgeSinkTask putItems call started: start={} attempts={} maxRetries={}",
           start,
           attempts,
           config.maxRetries);
+
+      remainingRecords =
+          eventBridgeWriter.putItems(remainingRecords).stream()
+              .filter(EventBridgeResult::isFailure)
+              .flatMap(it -> handleFailedEntries(it.failure(), attempts.get()))
+              .collect(toList());
+
+      // only report successful putEvents calls
+      statusReporter.setSentRecords(records.size());
+
       try {
-        batches.stream()
-            .flatMap(batch -> eventBridgeWriter.putItems(batch).stream())
-            .filter(EventBridgeWriterRecord::hasError)
-            .forEach(this::handleFailedEntries);
-        // only report successful putEvents calls
-        statusReporter.setSentRecords(eventBridgeWriterRecords.size());
-      } catch (EventBridgeWriterException e) {
-        switch (e.exceptionType) {
-          case RETRYABLE:
-            handleRetryableError(attempts, e);
-            continue; // retry
-          case NON_RETRYABLE:
-            handleNonRetryableError(e);
-        }
+        // TODO: improve retry delay implementation
+        TimeUnit.MILLISECONDS.sleep(config.retriesDelay);
+      } catch (InterruptedException ie) {
+        throw new ConnectException(ie);
       }
-      var completion = OffsetDateTime.now(ZoneOffset.UTC);
-      log.trace(
-          "EventBridgeSinkTask putItems call completed: start={} completion={} durationMillis={} "
-              + "attempts={} "
-              + "maxRetries={}",
-          start,
-          completion,
-          Duration.between(start, completion).toMillis(),
-          attempts,
-          config.maxRetries);
-      return;
     }
+
+    var completion = OffsetDateTime.now(ZoneOffset.UTC);
+    log.trace(
+        "EventBridgeSinkTask putItems call completed: start={} completion={} durationMillis={} "
+            + "attempts={} "
+            + "maxRetries={}",
+        start,
+        completion,
+        Duration.between(start, completion).toMillis(),
+        attempts,
+        config.maxRetries);
+  }
+
+  private Stream<SinkRecord> handleFailedEntries(
+      MappedSinkRecord<EventBridgeResult.Error> errorRecord, Integer attempts) {
+    switch (errorRecord.getValue().getLevel()) {
```

Comment:
nit: needs some comments

Reply:
As already comment above, the error structure is an intermediate solution. Will add comments if required on the final implementation.

Reply:
The latest commit refines the error handling @embano1.